### PR TITLE
LKE - Add support for pool tags and custom disk partitions.

### PR DESCRIPTION
**Pull request by kekoav**
For LKE, adds pool tags and custom disk partitions. This support is already present in the golinode library and API, just exposing it for Terraform.

### CODE REVIEW

File: linode/lke/cluster.go [hunk 1]
```diff
@@ -338,6 +338,14 @@ func waitGroupCh(wg *sync.WaitGroup) <-chan struct{} {
 	return done
 }
 
+func poolMatchesDeclaredPool(pool *linodego.LKEClusterPool, declaredPool map[string]interface{}) bool {
+	if pool.Type != declaredPool["type"] || pool.Count != declaredPool["count"] {
+		return false
+	}
+
+	return true
+}
```
Comment (LBGarber) [inline linode/lke/cluster.go:347]: Just a small stylistic change :+1: 
```suggestion
return pool.Type == declaredPool["type"] && pool.Count == declaredPool["count"]
```

**Review by LBGarber — CHANGES_REQUESTED** [review]
Hey there! Thanks for the great contribution :smile: 

Just a few small changes and we should be good to go. Do you mind if I push a commit to this PR shortly to add integration tests?

File: website/docs/r/lke_cluster.html.md [hunk 2]
```diff
@@ -75,6 +75,11 @@ The following arguments are supported in the `pool` specification block:
 
 * [`autoscaler`](#autoscaler) - (Optional) If defined, an autoscaler will be enabled with the given configuration.
 
+* `tags` - (Optional) An array of tags applied to the Node Pool. Tags are for organizational purposes only.
+
+* [`disk`](#disk) - (Optional) An array of custom partitions to create on each node, in addition to the boot partition.
+
+
```
Comment (LBGarber) [inline website/docs/r/lke_cluster.html.md:82]: This should fix the docs linting workflow
```suggestion
```

File: website/docs/d/lke_cluster.html.md [hunk 3]
```diff
@@ -62,4 +62,12 @@ In addition to all arguments above, the following attributes are exported:
 
     * `max` - The maximum number of nodes to autoscale to.
 
+  * `tags` - The tags applied to the node pool.
+
+  * `disk` - The custom disk partitions for the node pool.
+  
+    * `size` - The size in MB of the disk partition.
+
+    * `type` - The filesystem type of the partition. (`raw`, `ext4`)
```
Comment (LBGarber) [inline website/docs/d/lke_cluster.html.md:71]: Remove trailing space for docs linting
```suggestion
    * `type` - The filesystem type of the partition. (`raw`, `ext4`)
```

**Review by LBGarber — CHANGES_REQUESTED** [review]
Just a few changes to get the docs workflow passing

**Comment by kekoav** [comment]
@LBGarber I don't mind at all! It's all yours, your changes look good to me.

**Comment by kekoav** [comment]
@LBGarber I was trying to add some integration tests, but it turned out it was going to be more than I bargained for... Adding Tags and Disk to `ClusterPoolSpec` no longer made it suitable for use as a map key which it was in several places, and I just didn't have the time to dig into how all that worked.

**Comment by LBGarber** [comment]
While writing the integration tests I ran into a few issues with how update logic is handled. I'll see what I can do about this and get back to you! :+1:

**Comment by LBGarber** [comment]
@kekoav Thanks for the feedback! I definitely agree with you there, a full cluster recreation should only be a last resort and shouldn't be triggered by a node pool update. I think it would vastly simplify the update logic here to have node pools be their own resource, but this currently doesn't mesh well with the existing LKE API architecture. I'll see what I can do about implementing disks into the existing node pool update logic.
 
> Let me know how I can help, I don't want to get in the way but would be happy to provide use cases from the perspective of a typical Kubernetes user.

Feedback is always appreciated! If you have any ideas or suggestions, feel free to let me know :slightly_smiling_face:

**Comment by LBGarber** [comment]
Sorry about that, looks like the stale workflow didn't respect the `keep` tag.

**Comment by akoidan** [comment]
Is there anything preventing reopening and merging this PR? I'm facing the same issue, unable to get nodes of a newly created cluster described [here](https://github.com/linode/terraform-provider-linode/issues/595#issuecomment-1566092555)

**Pull request closed without merging**